src/gd_heif.c

```diff
@@ -23,13 +23,52 @@
 #define GD_HEIF_ALLOC_STEP (4*1024)
 #define GD_HEIF_HEADER 12
 
+static int heifInit = 0;
```

Comment:
bools should be bool, not int

Reply:
Maybe more importantly, that won't work for multi-threading, I think.

Reply:
neither type works for multithreading.  they'd need to be atomic and/or grab pthread locks.

that said, I don't think we've ever thought about MT, so I wouldn't be surprised if there were already issues in the codebase.

Reply:
I'm not aware of any multi-threading issues in libgd when used with PHP. :)

Reply:
gdkanji.c is not MT safe.  none of the error handler APIs are MT safe (which is to say, the handlers are per-process, not per-thread).  I guess most MT issues after that would be library-specific and not really on gd's plate.

maybe along those lines, should we punt these problems completely onto the user ?  if the only thing `gdHeifInit` does is call `heif_init`, and the only reason people would call `gdHeifInit` is because they want to explicitly use the HEIF format, then why not force them to call `heif_init` themselves ?  that would also let them call `heif_deinit` if they really wanted to.

if we can't handle it automatically in a format/library agnostic way, i don't think we're bringing any value-add here.